extensions/dim-rename/src/main/java/io/druid/query/extraction/ClusterDimRenameFn.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Druid - a distributed column store.
+ * Copyright 2012 - 2015 Metamarkets Group Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.druid.query.extraction;
+
+import com.fasterxml.jackson.annotation.JacksonInject;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.google.common.base.Function;
+import com.metamx.common.StringUtils;
+
+import javax.annotation.Nullable;
+import javax.validation.constraints.NotNull;
+import java.nio.ByteBuffer;
+import java.util.Map;
+
+/**
+ *
+ */
+@JsonTypeName("rename")
+public class ClusterDimRenameFn extends AbstractDimRenameFn implements DimExtractionFn
+{
+  private static final byte CACHE_KEY = 0x07;
+
+  private final String topic;
+
+  @JsonCreator
+  public ClusterDimRenameFn(
+      @JacksonInject final RenameManager renameManager,
```

Comment:
That's a good point as well...  Fwiw, I think that these pains are just pointing out that the initial "do the entire query API through JSON/Jackson" idea might not be right anymore.

Perhaps we should think about actually separating the query layer from the implementation layer a bit more?  We could explore more expressive or common languages (SQL, or whatever others are doing).    Then we could make a distinction between the things that just pass queries around from the things that actually process them?  (That's much bigger than this PR, but perhaps we should start thinking about it?)